service/s3/internal/customizations/updateEndpoint.go

```diff
@@ -0,0 +1,156 @@
+package customizations
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/awslabs/smithy-go"
+	"github.com/awslabs/smithy-go/middleware"
+	"github.com/awslabs/smithy-go/transport/http"
+)
+
+// UpdateEndpointOptions provides the options for the UpdateEndpoint middleware setup.
+type UpdateEndpointOptions struct {
+	UsePathStyle  bool
+	UseAccelerate bool
+
+	Region string
+}
+
+// UpdateEndpoint adds the middleware to the middleware stack based on the UpdateEndpointOptions.
+func UpdateEndpoint(stack *middleware.Stack, options UpdateEndpointOptions) {
```

Comment:
Instead of this middleware filing the endpoint directly, maybe this middleware should be generated as the  endpoint resolver, instead of the default one for other API clients.

Reply:
We may switch to this when we do other customization, but for now this stays a little ambiguous on how we would modify the endpoint with the multiple customization's in place.